PrestaShop/PrestaShop
# Improve extendability of the new product page form

| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.0.x
| Description?      | The new product page needs to be easily extendable by modules, so far we could already add some custom fields anywhere in the existing tabs but not add our own custom tab<br><br>To allow this a new `NavigationTabType` was introduced it basically allows you to display its children in a tabbed navigation each child represents a Tab This component is used as a base for the product form<br><br>Also in order to limit the twig content and especially the association of form theme a `FormThemeExtension` which introduces a new `form_theme` option that allows you to customize the form theme from the PHP form type code It's used on the whole product page which uses many sub form themes, but it will also allow modules to use their own form theme for their fields, or to customize the Core form theme if they need to<br><br>Finally for backward compatibility a custom `ExtraModulesType` form type was introduced to handle the integration of the `displayAdminProductsExtra` hook used in modules
| Type?             | new feature
| Category?         | BO
| BC breaks?        | yes
| Deprecations?     | no
| Fixed ticket?     | Fix #28994
| Related PRs       | You should use this PR for the example module https://github.com/PrestaShop/example-modules/pull/50
| How to test?      | See How to test section below
| Possible impacts? | IDs may have changed on some elements which could break some JS behaviour because the selectors wouldn't match anymore, that's why it'd be good to check that all fields work fine but not in a extensive way ^^ One edit, save, it's updated it's ok!





### How to test
This PR changes the structure of the form a bit, so we need to check that the form behaves like before and that all the fields work as expected The idea is **not to check everything extensively** but only a quick run on each input to make sure they are not ignored because of some naming mismatch just to be sure Simply edit the field, save, the field is saved, done!

Also to ensure the extendability of modules works fine, this is the actual purpose of this PR, you should use the [example module `demoproductform`](https://github.com/PrestaShop/example-modules/pull/50) and install it using the branch of this PR. Once it is installed it should add three new custom things in the product page:

#### Custom field in description tab

![Capture d’écran 2022-06-30 à 16 21 16](https://user-images.githubusercontent.com/13801017/176701548-1623f3d4-2ae5-4c6b-b2e0-53c17eaa468c.png)

#### Custom tab with a custom price field

![Capture d’écran 2022-06-30 à 16 21 54](https://user-images.githubusercontent.com/13801017/176701694-86131167-0877-441e-857c-9dd0425c7411.png)

#### Configuration panel in the Modules tab

It is nearly a resume of the custom field used as a POC to show the feature is still usable for backward compatibility with modules relying on the `displayAdminProductsExtra` hook.

![Capture d’écran 2022-06-30 à 16 22 32](https://user-images.githubusercontent.com/13801017/176701831-58660057-70de-4088-8242-337fe4b08c88.png)

Both fields should be editable and persisted in the DB, they are not used in FO right now only in the BO. If this PR is validated with the example module then it will also validate the module PR so we can merge it to provide an example for the community.

### Finally, as all the PR for the product page V2 this PR MUST NOT be tested for multishop.
